### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by defido**

### CODE REVIEW

**Comment by defido** [comment]
I have added some changes that make it a bit more clear how to get started and changed some naming to be a bit clearer. 

Also, started to write a bit more on tutorial side for the readme. 

Also, added more setup defaults to the env file and added some examples such as network setups and argument setups.

File: README.md [hunk 1]
```diff
@@ -18,3 +18,25 @@ After having installed Yarn, simply run:
...
 yarn install
 ```
+
+## Running deployments 
+
+```
+npx hardhat xdeploy
+```
+
+This starter use [xdeploy](https://github.com/pcaversaccio/xdeployer) check the documentation for more on deployments.
```
Comment (pcaversaccio) [inline README.md:28]:
```suggestion
This template uses the [xdeploy](https://github.com/pcaversaccio/xdeployer) Hardhat plugin. Check out the documentation for more information on the specifics of the deployments.
```

File: README.md [hunk 2]
```diff
@@ -18,3 +18,25 @@ After having installed Yarn, simply run:
...
+
+## env file
```
Comment (pcaversaccio) [inline README.md:30]:
```suggestion
## `.env` File 
```

File: README.md [hunk 3]
```diff
@@ -18,3 +18,25 @@ After having installed Yarn, simply run:
...
+
+```
+npx hardhat xdeploy
+```
+
+This starter use [xdeploy](https://github.com/pcaversaccio/xdeployer) check the documentation for more on deployments.
+
+## env file 
+
+In the .env file, place your wallets private key in the PRIVATE_KEY section, this allows access to your wallet to use both testnet and mainnet funds.
```
Comment (pcaversaccio) [inline README.md:32]:
```suggestion
In the `.env` file place the private key of your wallet in the `PRIVATE_KEY` section. This allows secure access to your wallet to use with both testnet and mainnet funds during Hardhat deployments. For more information on how this works, please read the documentation of the `npm` package [`dotenv`](https://www.npmjs.com/package/dotenv).
```

File: README.md [hunk 4]
```diff
@@ -18,3 +18,25 @@ After having installed Yarn, simply run:
...
+
+## Verification
+
+Change the contract address to your contract after deployment has been successful. This works for both testnet & mainnet. You will need to get an API key from etherscan.io, bscscan, snowtrace etc. 
+
+Typescript deployment example:
+npx hardhat verify --network fantomMain YOUR_CONTRACT_HERE --constructor-args deploy-arg.ts 
+
+Javascript deployment args example:
```
Comment (pcaversaccio) [inline README.md:41]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -18,3 +18,25 @@ After having installed Yarn, simply run:
...
+
+This starter use [xdeploy](https://github.com/pcaversaccio/xdeployer) check the documentation for more on deployments.
+
+## env file 
+
+In the .env file, place your wallets private key in the PRIVATE_KEY section, this allows access to your wallet to use both testnet and mainnet funds.
+
+## Verification
+
+Change the contract address to your contract after deployment has been successful. This works for both testnet & mainnet. You will need to get an API key from etherscan.io, bscscan, snowtrace etc.
```
Comment (pcaversaccio) [inline README.md:36]:
```suggestion
Change the contract address to your contract after the deployment has been successful. This works for both testnet and mainnet. You will need to get an API key from [etherscan](https://etherscan.io), [snowtrace](https://snowtrace.io) etc.
```

File: README.md [hunk 6]
```diff
@@ -18,3 +18,25 @@ After having installed Yarn, simply run:
...
+
+## env file 
+
+In the .env file, place your wallets private key in the PRIVATE_KEY section, this allows access to your wallet to use both testnet and mainnet funds.
+
+## Verification
+
+Change the contract address to your contract after deployment has been successful. This works for both testnet & mainnet. You will need to get an API key from etherscan.io, bscscan, snowtrace etc. 
+
+Typescript deployment example:
```
Comment (pcaversaccio) [inline README.md:38]:
```suggestion
**Example:**
```

File: README.md [hunk 7]
```diff
@@ -18,3 +18,25 @@ After having installed Yarn, simply run:
...
+npx hardhat verify --network fantomMain YOUR_CONTRACT_HERE --constructor-args deploy-arg.ts
```
Comment (pcaversaccio) [inline README.md:39]:
```suggestion
```ts
npx hardhat verify --network fantomMain <YOUR_CONTRACT_ADDRESS> --constructor-args deploy-arg.ts
```
```

File: deploy-args.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+// You may also use deployment arguments in JS if needed
```
Comment (pcaversaccio) [inline deploy-args.js:1]: please delete this file -> only TypeScript for this template!

File: deploy-args.ts [hunk 9]
```diff
@@ -1,2 +1,8 @@
-const data = ["Hello, Hardhat!"];
+// Input the arguments for the constructor
+const data = [
+  "Hello, Hardhat!", // The example Greeter.sol file needs a string
+];
+// Export the args to be picked up by the hardhat.config.ts deployment script
```
Comment (pcaversaccio) [inline deploy-args.ts:5]:
```suggestion
// Export the arguments to be picked up by the `hardhat.config.ts` deployment script
```

File: hardhat.config.ts [hunk 10]
```diff
@@ -38,6 +38,9 @@ task(
   }
 );
 
+
+// Input the solidity version that you are using in your make .sol file
```
Comment (pcaversaccio) [inline hardhat.config.ts:42]:
```suggestion
// Input the Solidity version you use in your files in the folder `contracts`
```

File: hardhat.config.ts [hunk 11]
```diff
@@ -106,99 +109,99 @@ const config: HardhatUserConfig = {
...
   },
   xdeploy: {
-    contract: "Greeter",
-    constructorArgsPath: "./deploy-args.ts",
-    salt: "WAGMI",
-    signer: process.env.PRIVATE_KEY,
-    networks: ["hardhat", "rinkeby", "kovan"],
-    rpcUrls: ["hardhat", process.env.RINKEBY_URL, process.env.KOVAN_URL],
-    gasLimit: 1.2 * 10 ** 6,
+    contract: "Greeter", // Change this name to the name of your main .sol file, "Token.sol" would be "Token"
```
Comment (pcaversaccio) [inline hardhat.config.ts:198]:
```suggestion
    contract: "Greeter", // Change this name to the name of your main .sol file; e.g. "Token.sol" would be "Token"
```

File: hardhat.config.ts [hunk 12]
```diff
@@ -106,99 +109,99 @@ const config: HardhatUserConfig = {
...
+    constructorArgsPath: "./deploy-args.ts", // Change to undefined if your constructor does not have any input arguments
+    salt: "WAGMI", // The salt must be the same for each chain you want to have a single contract address on. Change the salt if you are doing a re-deployment of the same codebase.
```
Comment (pcaversaccio) [inline hardhat.config.ts:200]:
```suggestion
    salt: "WAGMI", // The salt must be the same for each EVM chain for which you want to have a single contract address. Change the salt if you are doing a re-deployment with the same codebase
```

File: hardhat.config.ts [hunk 13]
```diff
@@ -106,99 +109,99 @@ const config: HardhatUserConfig = {
...
+    signer: process.env.PRIVATE_KEY, // This is your metamask or other wallet private key
```
Comment (pcaversaccio) [inline hardhat.config.ts:201]:
```suggestion
    signer: process.env.PRIVATE_KEY, // This is your wallet's private key
```

File: hardhat.config.ts [hunk 14]
```diff
@@ -106,99 +109,99 @@ const config: HardhatUserConfig = {
...
+    networks: ["hardhat", "rinkeby", "bscTestnet"], // Use network name found above ex: 'fantomTestnet', 'bscTestnet'. Use localhost or hardhat for local testing
```
Comment (pcaversaccio) [inline hardhat.config.ts:202]:
```suggestion
    networks: ["hardhat", "rinkeby", "bscTestnet"], // Use the network names specified here: https://github.com/pcaversaccio/xdeployer#configuration. Use `localhost` or `hardhat` for local testing
```

File: hardhat.config.ts [hunk 15]
```diff
@@ -106,99 +109,99 @@ const config: HardhatUserConfig = {
...
+    rpcUrls: ["hardhat", process.env.ETH_RINKEBY_TESTNET_URL, process.env.BSC_TESTNET_URL], // Use the matching ENV URL with your chosen RPC in the env file
```
Comment (pcaversaccio) [inline hardhat.config.ts:203]:
```suggestion
    rpcUrls: ["hardhat", process.env.ETH_RINKEBY_TESTNET_URL, process.env.BSC_TESTNET_URL], // Use the matching env URL with your chosen RPC in the `.env` file
```

File: hardhat.config.ts [hunk 16]
```diff
@@ -106,99 +109,99 @@ const config: HardhatUserConfig = {
...
+    gasLimit: 1.2 * 10 ** 6, // Maximum limit is 15 * 10 ** 6 or 15,000,000. If deployments are failing, try increasing this number
```
Comment (pcaversaccio) [inline hardhat.config.ts:204]:
```suggestion
    gasLimit: 1.2 * 10 ** 6, // Maximum limit is 15 * 10 ** 6 or 15,000,000. If the deployments are failing, try increasing this number. However, keep in mind that this costs money in a production environment!
```

**Review by pcaversaccio — APPROVED** [review]

**Pull request merged**